File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription
  (:require [clojure.string :as str]))

(def dnaToRnaMap (hash-map
  "G" "C"
  "C" "G"
  "T" "A"
  "A" "U"
))

(defn mapDnaToRna [dna-nucleotide]
  (assert (contains? dnaToRnaMap dna-nucleotide))
  (get dnaToRnaMap dna-nucleotide)
)

(defn to-rna [dna-strand]
  (str/join (for [dna-nucleotide (str/split dna-strand #"")] (mapDnaToRna dna-nucleotide)))
)

File: typescript/rna-transcription/rna-transcription.ts
enum dnaToRnaMap {
  G = "C",
  C = "G",
  T = "A",
  A = "U",
}

type dna = keyof typeof dnaToRnaMap;

export function toRna(strand: string): string {
  return strand
    .split("")
    .map((nucleotide) => {
      const rna = dnaToRnaMap[nucleotide as dna];
      if (!rna) throw new Error("Invalid input DNA.");
      return rna;
    })
    .join("");
}
